modAionImpl/src/org/aion/zero/impl/sync/handler/BroadcastTxHandler.java

```diff
@@ -77,22 +114,31 @@ public final void receive(int _nodeIdHashcode, String _displayId, final byte[] _
         List<byte[]> broadCastTx = BroadcastTx.decode(_msgBytes);
 
         if (broadCastTx == null) {
-            log.error("<broadcast-tx decode-error unable to decode tx-list from {}, len: {]>", _displayId, _msgBytes.length);
+            log.error("<BroadcastTxHandler decode-error unable to decode tx-list from {}, len: {]>", _displayId, _msgBytes.length);
             if (log.isTraceEnabled()) {
-                log.trace("broadcast-tx dump: {}", ByteUtil.toHexString(_msgBytes));
+                log.trace("BroadcastTxHandler dump: {}", ByteUtil.toHexString(_msgBytes));
             }
         }
 
         if (broadCastTx.isEmpty()) {
             p2pMgr.errCheck(_nodeIdHashcode, _displayId);
 
             if (log.isTraceEnabled()) {
-                log.trace("<broadcast-tx from: {} empty {}>", _displayId);
+                log.trace("<BroadcastTxHandler from: {} empty {}>", _displayId);
 
             }
             return;
         }
-        pendingState.addPendingTransactions(castRawTx(broadCastTx));
+
+        if (this.buffer) {
+            try {
+                txQueue.addAll(castRawTx(broadCastTx));
```

Comment:
Suggest changing to non-blocking add

Reply:
add and offer is same inside blocking method implemented by the blocking queue

    public boolean add(E e) {
        if (offer(e))
            return true;
        else
            throw new IllegalStateException("Queue full");
    }